### ISSUE TITLE
No documentation

### ISSUE THREAD

**Issue by praxidike97** [emojicode/allegro#1]
The link to the documentation on the emojicode site is broken. Is the doc available somewhere else?

**Comment by thbwd** [comment]
Thanks for noting that. I’m not sure the package works with the latest version anyway, that’s probably the reason why the documentation is not available at the moment. 

The source code contains all documentation in documentation comments though. If it works for you, let me know and I’ll make sure to restore the documentation.

**Comment by praxidike97** [comment]
Hey, thanks for the quick reply! This question might be stupid, but how do I install this package in the first place? I looked at the packages site using the Wayback Machine and it seems before it has been included per default in emojicode, so that it can simply be imported using 📦 allegro 🔴. But how can I use it now?

**Comment by thbwd** [comment]
It can be installed with yarn. The documentation has a guide on using yarn with Emojicode

**Comment by thbwd** [comment]
Of course, you should make sure you’ve installed the allegro libraries first (using your package manager or brew on mac)

**Comment by praxidike97** [comment]
Ok, tried to install allegro using yarn with:

yarn add allegro. 

But it installs a wrong library:

https://yarnpkg.com/en/package/allegro

Is allegro still available on yarn, I cannot find it?

**Comment by thbwd** [comment]
You need to provide the git URL:

```
yarn add git@github.com:emojicode/allegro.git
```

Hope that works.

**Comment by praxidike97** [comment]
Ok, I managed to get the allegro module from GitHub, but I still have a problem. When I am want to import it into my own project I get the error message

🚨 error: Couldn't read input file /usr/local/EmojicodePackages/allegro/interface.emojii.

(This file also does not exist)

Therefore, I tried to compile the .emojic file. When I run "emojicodec -p allegro main.emojic" I get

main.emojic:56:6: 🚨 error: Expected BlockBegin but instead found Error(🚨).
  🆕 📄🚨🌩  path 🔡 📻 🔤allegroNewBitmapPath🔤
     ⬆️

and when I run "emojicodec -p allegro app.emojic" all I get is 

app.emojic:20:15: 🚨 error: Could not find a type named 🖼 in namespace 🏠.
  ❗️ 🏓 bitmap 🖼 📻 🔤allegroAppSetTargetBitmap🔤

I am a little bit lost, what am I supposed to do now?

**Comment by praxidike97** [comment]
Ah, I forgot to mention that when I add allegro from yarn I get the following message:


yarn add v1.16.0
info No lockfile found.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
error /usr/local/EmojicodePackages/node_modules/allegro: Command failed.
Exit code: 1
Command: cmake . && make
Arguments: 
Directory: /usr/local/EmojicodePackages/node_modules/allegro
Output:
-- The C compiler identification is AppleClang 10.0.0.10001145
-- The CXX compiler identification is AppleClang 10.0.0.10001145
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
SYSTEM_ALLEGRO
    linked by target "allegro" in directory /usr/local/EmojicodePackages/node_modules/allegro

-- Configuring incomplete, errors occurred!
See also "/usr/local/EmojicodePackages/node_modules/allegro/CMakeFiles/CMakeOutput.log".
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.

Seems like something is going wrong?

**Comment by thbwd** [comment]
The errors you get when compiling yourself tell me that this package is not compatible with Emojicode 0.9. This is probably just due to minor syntax changes, for instance `🚨🌩` is no longer used in declarations. 

Have you installed the allegro development libraries? That’s the reason for the build error message. Furthermore, you have not configured yarn to use the right directory as described in the guide.

**Comment by thbwd** [comment]
As for the compatiblity issue, maybe you can fix those yourself. Otherwise I’ll try to address this issue for the latest 1.0 pre-release, which I hope to release within the next week or so.

**Comment by praxidike97** [comment]
Hm, I am not sure I know what you mean. What are the "allegro development libraries"? And what directory should be configured? /usr/local/EmojicodePackages is not the right place for new packages?

Btw, next week will be release of version 1.0?? Awesome!

**Comment by thbwd** [comment]
Sorry for the delay. I have fixed the compatibility issues and tested it with 1.0. 

Usage example:

```
mkdir allegro_test
cd allegro_test/
echo "--modules-folder packages" > .yarnrc
yarn init
yarn add https://github.com/emojicode/allegro
nano allegro.emojic # create your test file
emojicodec allegro.emojic
./allegro
```

You could install the package into /usr/local/EmojicodePackages but I wouldn’t recommend that.

The following shows the basic usage of the package:
https://gist.github.com/thbwd/ab7e7014e8afaf2f9d2ba0a44140f870 

There are still some minor issues and I’ll add the package back to the index after those are fixed.
Hope this helps

**Comment by thbwd** [comment]
"allegro development libraries" are either

- `brew install allegro` (macOS)
- `sudo apt install liballegro5-dev liballegro-acodec5-dev liballegro-audio5-dev \
    liballegro-image5-dev liballegro-dialog5-dev liballegro-ttf5-dev \
    liballegro-physfs5-dev` (Debian, Ubuntu, etc.)
- or whatever package manager you use
